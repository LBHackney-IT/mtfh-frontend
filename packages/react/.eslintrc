{
  "extends": "@lbh-mtfh/eslint-config",
  "parserOptions": {
    "project": "tsconfig.json"
  },
  "rules": {
    "prefer-arrow-callback": "off",
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/*/rollup.config.js"]
      }
    ],
    "jsx-a11y/anchor-is-valid": [
      "error",
      {
        "components": ["Link"],
        "aspects": ["invalidHref"]
      }
    ]
  }
}
