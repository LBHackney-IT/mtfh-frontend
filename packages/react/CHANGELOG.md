# @lbh-mtfh/react

## 1.1.0

### Minor Changes

- 37b7b3e: Initial release

### Patch Changes

- Updated dependencies [37b7b3e]
  - @lbh-mtfh/system@1.1.0
  - @lbh-mtfh/test-utils@1.1.0
