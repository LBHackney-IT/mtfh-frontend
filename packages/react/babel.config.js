const babel = require("../../babel.config")

module.exports = babel
