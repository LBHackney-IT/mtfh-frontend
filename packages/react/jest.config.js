const baseConfig = require("../../jest.config")

baseConfig.setupFilesAfterEnv.push("<rootDir>/jest.setup.ts")

module.exports = {
  ...baseConfig,
}
