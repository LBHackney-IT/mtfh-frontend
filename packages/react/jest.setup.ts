import * as LBHFrontend from "lbh-frontend"

window.LBHFrontend = LBHFrontend
