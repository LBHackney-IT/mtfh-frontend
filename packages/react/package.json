{
  "name": "@lbh-mtfh/react",
  "version": "1.0.0",
  "description": "React component library of the LBH Design System",
  "author": "https://hackney.gov.uk/",
  "homepage": "https://github.com/LBHackney-IT/mtfh-frontend",
  "license": "MIT",
  "sideEffects": false,
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/esm/index.d.ts",
  "exports": {
    ".": {
      "require": "./dist/cjs/index.js",
      "default": "./dist/esm/index.js"
    }
  },
  "files": [
    "dist"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/LBHackney-IT/mtfh-frontend.git"
  },
  "scripts": {
    "test": "jest --env=jsdom --passWithNoTests",
    "build": "rollup -c"
  },
  "bugs": {
    "url": "https://github.com/LBHackney-IT/mtfh-frontend/issues"
  },
  "devDependencies": {
    "@lbh-mtfh/eslint-config": "^1.0.0",
    "@rollup/plugin-commonjs": "^19.0.0",
    "@rollup/plugin-node-resolve": "^13.0.0",
    "@rollup/plugin-typescript": "^8.2.1",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@testing-library/user-event": "^13.1.9",
    "axe-core": "^4.2.3",
    "axios": "^0.21.1",
    "jest": "^27.0.6",
    "jest-axe": "^5.0.1",
    "node-sass": "^6.0.1",
    "postcss": "^8.3.5",
    "rollup": "^2.52.8",
    "rollup-plugin-peer-deps-external": "^2.2.4",
    "rollup-plugin-postcss": "^4.0.0"
  },
  "dependencies": {
    "@lbh-mtfh/system": "^1.0.0",
    "@lbh-mtfh/test-utils": "^1.0.0",
    "@mtfh/common": "https://github.com/LBHackney-IT/mtfh-frontend-common.git",
    "@radix-ui/react-polymorphic": "^0.0.12",
    "@reach/accordion": "^0.15.3",
    "@reach/dialog": "^0.15.3",
    "@types/lbh-frontend": "^1.0.0",
    "classnames": "^2.3.1",
    "generic-names": "^3.0.0",
    "govuk-frontend": "^3.13.0",
    "lbh-frontend": "^3.5.3",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-merge-refs": "^1.1.0",
    "swr": "^0.5.6",
    "use-breakpoint": "^2.0.1"
  },
  "peerDependencies": {
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  }
}
