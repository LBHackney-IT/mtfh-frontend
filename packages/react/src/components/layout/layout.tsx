import React, { ComponentPropsWithoutRef, ReactElement, forwardRef } from "react";
import cn from "classnames";

import "./styles.module.scss";

export interface LayoutProps extends ComponentPropsWithoutRef<"div"> {
  top?: ReactElement;
  side?: ReactElement;
}

export const Layout = forwardRef<HTMLDivElement, LayoutProps>(function Layout(
  { children, top, side, className, ...props },
  ref,
) {
  return (
    <div
      ref={ref}
      className={cn("mtfh-layout", { "mtfh-layout--narrow": !side }, className)}
      {...props}
    >
      {top}
      <div className="mtfh-layout__container">
        {side ? <div className="mtfh-layout__aside">{side}</div> : null}
        <div className="mtfh-layout__main">{children}</div>
      </div>
    </div>
  );
});
