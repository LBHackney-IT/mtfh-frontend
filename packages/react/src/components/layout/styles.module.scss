@import "~lbh-frontend/lbh/base";
@import "~govuk-frontend/govuk/helpers/media-queries";

.layout {
  @include govuk-media-query($from: desktop) {
    &__container {
      display: grid;
      grid-row-gap: 0;
      grid-column-gap: 30px;
      grid-template-rows: 1fr;
      grid-template-columns: 310px 1fr;
    }

    &--narrow .layout__container {
      grid-template-columns: 1fr;
    }

    &__aside {
      grid-area: 1 / 1 / 2 / 2;
    }

    &__main {
      margin-top: 0;

      .layout:not(.layout--narrow) & {
        grid-area: 1 / 2 / 2 / 3;
      }
    }
  }
}
