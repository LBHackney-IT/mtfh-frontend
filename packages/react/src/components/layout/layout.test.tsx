import React from "react"
import { testA11y } from "@hackney/mtfh-test-utils"
import { render } from "@testing-library/react"

import { Layout } from "./layout"

test("it renders correctly", async () => {
  const { container } = render(
    <Layout side={<div id="side">Side Bar</div>} top={<div id="top">Top</div>}>
      <div id="main">Main</div>
    </Layout>,
  )
  expect(container).toMatchSnapshot()
  await testA11y(container)
})

test("it renders correctly without a side", () => {
  const { container } = render(
    <Layout>
      <div id="main">Main</div>
    </Layout>,
  )
  expect(container).toMatchSnapshot()
})
