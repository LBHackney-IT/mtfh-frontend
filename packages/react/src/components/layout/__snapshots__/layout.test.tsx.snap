// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div>
  <div
    class="mtfh-layout"
  >
    <div
      id="top"
    >
      Top
    </div>
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__aside"
      >
        <div
          id="side"
        >
          Side Bar
        </div>
      </div>
      <div
        class="mtfh-layout__main"
      >
        <div
          id="main"
        >
          Main
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`it renders correctly without a side 1`] = `
<div>
  <div
    class="mtfh-layout mtfh-layout--narrow"
  >
    <div
      class="mtfh-layout__container"
    >
      <div
        class="mtfh-layout__main"
      >
        <div
          id="main"
        >
          Main
        </div>
      </div>
    </div>
  </div>
</div>
`;
